tedana/workflows/tedana.py

```diff
@@ -499,15 +499,24 @@ def tedana_workflow(data, tes, out_dir='.', mask=None,
         RepLGR.info("An initial mask was generated from the first echo using "
                     "nilearn's compute_epi_mask function.")
 
-    # Create an adaptive mask with at least 3 good echoes.
-    mask, masksum = utils.make_adaptive_mask(catd, mask=mask, getsum=True, threshold=3)
-    LGR.debug('Retaining {}/{} samples'.format(mask.sum(), n_samp))
-    io_generator.save_file(masksum, "adaptive mask img")
+    # Create an adaptive mask with at least 1 good echo, for denoising
+    mask_denoise, masksum_denoise = utils.make_adaptive_mask(
+        catd,
+        mask=mask,
+        getsum=True,
+        threshold=1
+    )
+    LGR.debug('Retaining {}/{} samples for denoising'.format(mask_denoise.sum(), n_samp))
+    io_generator.save_file(masksum_denoise, "adaptive mask img")
+
+    # Create an adaptive mask with at least 3 good echoes, for classification
+    mask_clf, masksum_clf = utils.make_adaptive_mask(catd, mask=mask, getsum=True, threshold=3)
+    LGR.debug('Retaining {}/{} samples for denoising'.format(mask_clf.sum(), n_samp))
 
     if t2smap is None:
         LGR.info('Computing T2* map')
         t2s_limited, s0_limited, t2s_full, s0_full = decay.fit_decay(
```

Comment:
Given the change in the workflow, I'm thinking that we should probably write out `[t2s|s0]_full` as `T2starmap|S0map` and `[t2s|s0]_limited` as `desc-limited_[T2starmap|S0map]`. Thoughts?

Reply:
To clarify: we're basically switching which is the "default" T2starmap per naming convention, correct? As an example, fMRIPrep would now take the "full" T2* map as the default, rather than the "limited"? I'm of mixed feelings on this because it's technically more representative of what we're doing, but a backwards incompatibility failure would be silent.

Reply:
I don't think many people actually use the T2* map for anything, but I do agree that we are modifying the outputs. I don't think that fMRIPrep will use the T2* map, since T2*-based coregistration has been deprecated, so fMRIPrep will only use the optimally combined data. That said, those optimally combined data are _also_ updated since they use the liberal mask instead of the conservative one.

Reply:
That makes sense. I think we should go ahead and change the names, then, and just include a warning at the next release.